shared/bls/bls.go

```diff
@@ -0,0 +1,47 @@
+// Package bls implements a go-wrapper around a C BLS library leveraging
+// the BLS12-381 curve. This package exposes a public API for verifying and
+// aggregating BLS signatures used by Ethereum 2.0.
+package bls
+
+import "fmt"
+
+// Signature used in the BLS signature scheme.
+type Signature struct{}
+
+// SecretKey used in the BLS scheme.
+type SecretKey struct{}
+
+// PublicKey corresponding to secret key used in the BLS scheme.
+type PublicKey struct{}
+
```

Comment:
Shouldnt you have a function that derives a public key from its secret key ?
Maybe something like this:
```
func (s *SecretKey) PublicKey() *PublicKey
```